Implement Azure Timeseries Insights Connector (#1)
ballerina-platform/module-azuretimeseries: README.md

## Module Overview

The Azure Insights connector allows you to query Azure Time-series Insights through the REST API. It also allows you to access metadata on environments along with querying for events and aggregates. It handles OAuth 2.0 authentication.

## Azure Insights Operations

The `ballerinax/azure.timeseries` module allows you to perform the following operations.

- Get all available environment meta data
- Get environment availability
- Get environment metadata
- Query environment for events
- Query environment for aggregates

## Compatibility

|                             |       Versions              |
|:---------------------------:|:---------------------------:|

> | Ballerina Language          | 1.1.x                       |

Reviewer: ```suggestion
| Ballerina Language          | 1.2.x                       |
```

| Azure Time-series Insights API          | 2016-12-12                  |

## Feature Overview

### Azure Insights Clients

There are 2 clients provided by Ballerina to interact with different API groups of the Timeseries Insights REST API. 

1. **azure.timeseries:InsightsClient** - This client is the top-most client in the Timeseries module. 
This can be used to get the relevant metadata associated with the environments available to the user.

2. **azure.timeseries:EnvironmentClient** - This client can be used to get metadata or query a specific environment.

## Getting Started

### Prerequisites

> Download and install [Ballerina](https://ballerinalang.org/downloads/).

Reviewer: ```suggestion
Download and install [Ballerina](https://ballerina.io/downloads/).
```

### Pull the Module
Execute the below command to pull the `Azure.Timeseries` module from Ballerina Central:

```ballerina
$ ballerina pull ballerinax/azure.timeseries
```

## Sample

The EnvironmentClient `getEvents()` remote function can be used to query events available in the environment.

```ballerina
import ballerina/log;
import ballerinax/azure.timeseries;

> azure.timeseries:ConnectionConfiguration connConfig = {

Reviewer: ```suggestion
timeseries:ConnectionConfiguration connConfig = {
```

    tenantId: <tenant Id>,
    clientId: <client id>,
    clientSecrect: <client secrect>
};

public function main() {
    

>     azure.timeseries:EnvironmentClient environmentClient = new(<envFQDN>, connConfig);

Reviewer: ```suggestion
    timeseries:EnvironmentClient environmentClient = new(<envFQDN>, connConfig);
```

## Module Overview

> The Azure Insights connector allows you to query Azure Time-series Insights through the  REST API. It also allows you access metadata on environments along with querying for events and aggregates. It handles OAuth 2.0 authentication.

Reviewer 2: ```suggestion
The Azure Insights connector allows you to query Azure Time-series Insights through the REST API. It also allows you to access metadata on environments along with querying for events and aggregates. It handles OAuth 2.0 authentication.
```

Author: Fixed in cbf5566

## Azure Insights Operations

> The `ballerinax/azure.timeseries` module allows you to perform following operations,

Reviewer 2: ```suggestion
The `ballerinax/azure.timeseries` module allows you to perform the following operations.
```

Author: Fixed in cbf5566

- Get all available environment meta data

> - Get Environment availability

Reviewer 2: ```suggestion
- Get environment availability
```

Author: Fixed in cbf5566

> - Get Environment metadata

Reviewer 2: ```suggestion
- Get environment metadata
```

Author: Fixed in cbf5566

- Query environment for events
- Query environment for aggregates

## Compatibility

|                             |       Versions              |
|:---------------------------:|:---------------------------:|
| Ballerina Language          | 1.1.x                       |
| Azure Time-series Insights API          | 2016-12-12                  |

## Feature Overview

### Azure Insights Clients

There are 2 clients provided by Ballerina to interact with different API groups of the Timeseries Insights REST API. 

1. **azure.timeseries:InsightsClient** - This client is the top-most client in the Timeseries module. 
This can be used to get the relevant metadata associated with the environments available to the user.

> 2. **azure.timeseries:EnvironmentClient** - Client that can be used to get metadata/query a specific environment.

Reviewer 2: ```suggestion
2. **azure.timeseries:EnvironmentClient** - This client can be used to get metadata or query a specific environment.
```

Author: Fixed in cbf5566

## Getting Started

### Prerequisites
Download and install [Ballerina](https://ballerinalang.org/downloads/).

### Pull the Module

> Execute the below command to pull the Azure.Timeseries module from Ballerina Central:

Reviewer 2: ```suggestion
Execute the below command to pull the `Azure.Timeseries` module from Ballerina Central:
```

Author: Fixed in cbf5566
